android/src/main/java/com/stripeterminalreactnative/listener/RNReaderReconnectionListener.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripeterminalreactnative.listener
+
+import com.facebook.react.bridge.ReactApplicationContext
+import com.stripe.stripeterminal.external.callable.Cancelable
+import com.stripe.stripeterminal.external.callable.ReaderReconnectionListener
+import com.stripe.stripeterminal.external.models.Reader
+import com.stripeterminalreactnative.ReactExtensions.sendEvent
+import com.stripeterminalreactnative.ReactNativeConstants.*
+import com.stripeterminalreactnative.mapFromReader
+
+class RNReaderReconnectionListener(
+    private val context: ReactApplicationContext
+) : ReaderReconnectionListener {
+
+    override fun onReaderReconnectFailed(reader: Reader) {
+        context.sendEvent(TERMINAL_FAIL_READER_RECONNECT.listenerName) {
+            putMap("reader", mapFromReader(reader))
+        }
+    }
```

Comment:
do we need to put a failure message here similar to https://github.com/stripe/stripe-terminal-react-native/blob/1870117cc3e1e872d8a8b7eafbe6f191c2d93d93/android/src/main/java/com/stripeterminalreactnative/listener/RNTerminalListener.kt#L20-L24
cc @chr-stripe